Parsers/CiscoMeraki/CiscoMeraki.txt

```diff
@@ -0,0 +1,156 @@
+// 1380653443.857790533 MS220_8P events type=8021x_eap_success port='' identity='employee@ikarem.com'
+//
+//
+let LogHeader = Syslog
+| where Computer in ("server1", "server2") // server1 and server2 are examples, replace this list with your Cisco Meraki device(s)
+| extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$",dynamic([1,2,3,4]),SyslogMessage)
+| mv-expand Parser
+| extend Epoch = tostring(Parser[0]),
+        DeviceName = tostring(Parser[1]),
+        LogType = tostring(Parser[2]),
+        Substring = tostring(Parser[3])
+| extend EpochTimestamp = split(Epoch,".")
+| extend EventTimestamp = unixtime_seconds_todatetime(tolong(EpochTimestamp[0]))
+| project-away EpochTimestamp, Parser;
+let UrlEvents = LogHeader
+| where LogType == "urls"
+| extend SrcIpAddr = extract(@"src=([0-9\.]+)\:",1,Substring),
+        SrcPortNumber = toint(extract(@"src=([0-9\.]+)\:(\d+)\s",2,Substring)),
+        DstIpAddr = extract(@"dst=([0-9\.]+)\:",1,Substring),
+        DstPortNumber = toint(extract(@"dst=([0-9\.]+)\:(\d+)\s",2,Substring)),
+        HttpRequestMethod = extract(@"request: (\w+)\s",1,Substring),
+        Url = extract(@"request: (\w+)\s(\S+)",2,Substring)
+| project-away Substring;
+let FlowEvents = LogHeader
+| where LogType contains "flow"
+| extend SrcIpAddr = extract(@"src=([0-9\.]+)\s",1,Substring),
+        SrcPortNumber = toint(extract(@"sport=(\S+)",1,Substring)),
+        SrcMacAddr = tostring(extract(@"mac=(\S+)\s",1,Substring)),
+        DstIpAddr = extract(@"dst=([0-9\.]+)\s",1,Substring),
+        NetworkProtocol = extract(@"protocol=(\w+)\s",1,Substring),
+        DstPortNumber = toint(extract(@"dport=(\S+)",1,Substring)),
+        Pattern = extract(@"pattern\: ([\S\s]+)",1,Substring)
+| project-away Substring;
+let AirMarshallEvents = LogHeader
+| where LogType == "airmarshal_events"
+| extend EventType = tostring(extract(@"type=(\S+)",1,Substring)),
+        Ssid = tostring(extract(@"ssid=\'(\S+)\'",1,Substring)),
+        Vap = toint(extract(@"vap=\'(\S+)\'\s",1,Substring)),
+        Bssid = tostring(extract(@"bssid=\'(\S+)\'",1,Substring)),
+        SrcMacAddr = tostring(extract(@"src=\'(\S+)\'",1,Substring)),
+        DstMacAddr = tostring(extract(@"dst=\'(\S+)\'",1,Substring)),
+        WiredMacAddr = tostring(extract(@"wired_mac=\'(\S+)\'",1,Substring)),
+        Channel = toint(extract(@"channel=\'(\d+)\'",1,Substring)),
+        VlanId = toint(extract(@"vlan_id=\'(\d+)\'",1,Substring)),
+        Rssi = toint(extract(@"rssi=\'(\d+)\'",1,Substring)),
+        FcType = toint(extract(@"fc_type=\'(\S+)\'",1,Substring)),
+        FcSubType = toint(extract(@"fc_subtype=\'(\S+)\'",1,Substring))
+| project-away Substring;
+let SecurityEvents = LogHeader
```

Comment:
Can't use SecurityEvents here it will conflict with the default table in Azure Sentinel called the same name.  Maybe use "SecEvents" instead, just cannot be SecurityEvents. :)

Reply:
updates made